rct229/report_engine/rct_report.py

```diff
@@ -40,23 +40,29 @@ def generate(self, rct_outcome, report_dir):
 
         """
         invalid_msg = rct_outcome["invalid_rmrs"]
-        outcomes = rct_outcome["outcomes"]
+        # Sort the outcomes by id
+        outcomes = sorted(
+            # The key used below splits the id on "-" and ensures the outcomes are sorted by the first components of the ids, then the second components, and so on
+            rct_outcome["outcomes"],
+            key=lambda x: [int(y) for y in x["id"].split("-")],
+        )
         if invalid_msg:
             print(f"Invalid RMDs: {str(invalid_msg)}\n")
         else:
             ruleset_report = self.initialize_ruleset_report()
             for outcome in outcomes:
-                rule_outcome_dict = {
-                    RCTOutcomeLabel.PASS: 0,
-                    RCTOutcomeLabel.FAILED: 0,
-                    RCTOutcomeLabel.UNDETERMINED: 0,
-                    RCTOutcomeLabel.NOT_APPLICABLE: 0,
-                }
-                rule_report = self.generate_rule_report(outcome, rule_outcome_dict)
-                rule_outcome = self.calculate_rule_outcome(rule_outcome_dict)
-                self.add_rule_to_ruleset_report(
-                    ruleset_report, rule_report, rule_outcome
-                )
+                if outcome["primary_rule"]:  # output only primary rules
```

Comment:
Good one, thanks for adding this check

I suggest to add a TODO comment here.
The reason is we havent decided whether we want to add secondary rule outcome in the final report or not.

Reply:
Thanks Weili! I just put a ```TODO``` comment here.